### PR TITLE
add header to use secondary db in file api status and last modified calls (WP-672)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: src/File/FileApi.php [hunk 1]
```diff
@@ -60,12 +61,20 @@ protected function processBodyOptions($requestData = []) {
     }
 
     /**
-     * {@inheritdoc}
-     *
```
Comment (vsolovei-smartling) [inline src/File/FileApi.php:64]: Parent phpdoc empty, nothing to inherit

File: src/File/FileApi.php [hunk 2]
```diff
@@ -222,43 +231,30 @@ public function downloadMultipleTranslationsOfFiles(DownloadMultipleFilesParamet
...
+    public function getStatusForAllLocales($fileUri, ParameterInterface $params = null, $tolerateOutdated = true)
     {
-        $params = (\is_null($params)) ? [] : $params->exportToArray();
-        $params['fileUri'] = $fileUri;
-
-        $requestData = $this->getDefaultRequestData('query', $params);
-
-        return $this->sendRequest("/file/status", $requestData, self::HTTP_METHOD_GET);
+        return $this->getStatusAllLocales($fileUri, $params, $tolerateOutdated);
     }
```
Comment (vsolovei-smartling) [inline src/File/FileApi.php:258]: Methods were identical

**Review by dimitrystd — CHANGES_REQUESTED** [review]
Let's discuss other options. I'm not sure we would like to share this header in public sdk.

**Comment by vsolovei-smartling** [comment]
> Let's discuss other options. I'm not sure we would like to share this header in public sdk.

WordPress calls these api methods, so we're limited with possible solutions. I think the best way then would be to allow injecting custom http client to FileApi. Other alternatives I see are:

- make a private fork of the repo (somewhat increase the amount of maintenance when adding other WordPress related changes, but not by much, hopefully)
- copy api to WordPress repository (much greater initial effort)

**Comment by PavelLoparev** [comment]
Should we close this since it was resolved in another way?

**Comment by vsolovei-smartling** [comment]
Resolved in another way

**Pull request closed without merging**